src/managers/KeybindManager.cpp

```diff
@@ -838,12 +853,15 @@ void CKeybindManager::toggleActiveFloating(std::string args) {
     if (!PWINDOW)
         return;
 
-    // remove drag status
-    g_pInputManager->currentlyDraggedWindow = nullptr;
-
     if (PWINDOW->m_sGroupData.pNextWindow && PWINDOW->m_sGroupData.pNextWindow != PWINDOW) {
+        const auto PCURRENT = PWINDOW->getGroupCurrent();
+        if ((forceFloat == 1 && PCURRENT->m_bIsFloating) || (forceFloat == 2 && !PCURRENT->m_bIsFloating)) {
```

Comment:
a) no {} around short ifs
b) `forceFloat != 0 && forceFloat - 1 == !PCURRENT->m_bIsFloating`

Reply:
also why is this even in the if, can be extracted outside of it to avoid duplicating code

Reply:
It's there twice because it depends on either PWINDOW or PCURRENT, which is defined inside the outer if.

Reply:
You sure you want that shortened version?  Seems way less clear what the logic is.

Reply:
> It's there twice because it depends on either PWINDOW or PCURRENT, which is defined inside the outer if.

right. The other can be extracted, though.

Reply:
> You sure you want that shortened version? Seems way less clear what the logic is.

yea

Reply:
You mean this?

```// forceFloat: 0 for no force, 1 for force float, 2 for force tile
void toggleActiveFloatingCore(std::string args, unsigned char forceFloat) {
    CWindow* PWINDOW = nullptr;

    if (args != "active" && args.length() > 1)
        PWINDOW = g_pCompositor->getWindowByRegex(args);
    else
        PWINDOW = g_pCompositor->m_pLastWindow;

    if (!PWINDOW)
        return;

    if (forceFloat != 0 && forceFloat - 1 == !PWINDOW->m_bIsFloating)
        return;

    if (PWINDOW->m_sGroupData.pNextWindow && PWINDOW->m_sGroupData.pNextWindow != PWINDOW) {
        const auto PCURRENT = PWINDOW->getGroupCurrent();
        if (forceFloat != 0 && forceFloat - 1 == !PCURRENT->m_bIsFloating)
            return;

        // remove drag status
        g_pInputManager->currentlyDraggedWindow = nullptr;

        PCURRENT->m_bIsFloating = !PCURRENT->m_bIsFloating;
        g_pLayoutManager->getCurrentLayout()->changeWindowFloatingMode(PCURRENT);

        CWindow* curr = PCURRENT->m_sGroupData.pNextWindow;
        while (curr != PCURRENT) {
            curr->m_bIsFloating = PCURRENT->m_bIsFloating;
            curr->updateDynamicRules();
            curr->updateSpecialRenderData();
            curr = curr->m_sGroupData.pNextWindow;
        }
    } else {

        // remove drag status
        g_pInputManager->currentlyDraggedWindow = nullptr;

        PWINDOW->m_bIsFloating = !PWINDOW->m_bIsFloating;

        PWINDOW->updateDynamicRules();

        g_pLayoutManager->getCurrentLayout()->changeWindowFloatingMode(PWINDOW);
    }
}
```

it would then run 
```
    if (forceFloat != 0 && forceFloat - 1 == !PWINDOW->m_bIsFloating)
        return;
``` 
in both cases... is that correct?

Reply:
yeah tho tbh I don't see why the PCURRENT check is necessary, all windows in a group should have their floating state synced. You can remove the second check mostl ikely

Reply:
I hadn't tried the group functionality at all before just this moment, but the intention was to preserve the exact behavior that was there except allow for "make (whatever) floating unless it already is" and "make (whatever) tiled unless it already is".  I think that's how I had it?

Reply:
Oh I see.  So if you had toggled it already using the PCURRENT etc. logic, all of the windows in the group would have the same floating state, that's what you mean?

Reply:
```// forceFloat: 0 for no force, 1 for force float, 2 for force tile
void toggleActiveFloatingCore(std::string args, unsigned char forceFloat) {
    CWindow* PWINDOW = nullptr;

    if (args != "active" && args.length() > 1)
        PWINDOW = g_pCompositor->getWindowByRegex(args);
    else
        PWINDOW = g_pCompositor->m_pLastWindow;

    if (!PWINDOW)
        return;

    if (forceFloat != 0 && forceFloat - 1 == !PWINDOW->m_bIsFloating)
        return;

    // remove drag status
    g_pInputManager->currentlyDraggedWindow = nullptr;

    if (PWINDOW->m_sGroupData.pNextWindow && PWINDOW->m_sGroupData.pNextWindow != PWINDOW) {
        const auto PCURRENT = PWINDOW->getGroupCurrent();


        PCURRENT->m_bIsFloating = !PCURRENT->m_bIsFloating;
        g_pLayoutManager->getCurrentLayout()->changeWindowFloatingMode(PCURRENT);

        CWindow* curr = PCURRENT->m_sGroupData.pNextWindow;
        while (curr != PCURRENT) {
            curr->m_bIsFloating = PCURRENT->m_bIsFloating;
            curr->updateDynamicRules();
            curr->updateSpecialRenderData();
            curr = curr->m_sGroupData.pNextWindow;
        }
    } else {
        PWINDOW->m_bIsFloating = !PWINDOW->m_bIsFloating;

        PWINDOW->updateDynamicRules();

        g_pLayoutManager->getCurrentLayout()->changeWindowFloatingMode(PWINDOW);
    }
}

```

Reply:
yea lgtm